packages/typescript-to-proptypes/src/getPropTypesFromFile.ts

```diff
@@ -206,6 +248,38 @@ function checkType({
     return createLiteralType({ jsDoc, value: 'null' });
   }
 
+  if (type.flags & ts.TypeFlags.IndexedAccess) {
```

Comment:
The problem is that, for conditional types inside an `Omit`, TS does not resolve its value, it just says "it's looks like `A['key']`".

My ugly solution is to retrieve its resolved parent with `objectType`, and if this parent looks like `type A<T> = T extends B ? C : D`, then I resolve both sides of the ternary and for each I get the property value and create an union.

Reply:
IMHO, it seems only partially hacky.
We have a known case that we handle conditionally, it does not produce any side effects in the `Core` repo and none in the `X` repo—sounds like a solution to me. 🤔 🙈 💡

Reply:
Make sens for the generation of proptypes :+1:

I assume TS only keep the index pointing to it in order to easily manage the case where the condition get defined. Which we don't care for proptypes